js/index.js

```diff
@@ -64,61 +51,53 @@ function initBoardEvListener() {
                 const squareString = cols[j] + i.toString();
                 const element = document.getElementById(squareString);
                 element.addEventListener("click", () => {
-                    handleClick(squareString);
+                    handleSquareClick(squareString);
                 });
             }
         }
     }
 
 initBoardEvListener();
 
-function handleClick(squareString){
-    const piece = getPiece(squareString);
-    if (piece && piece.getPlayerId() == currentPlayer) {
-        activePiece = piece;
-        colorAllSquares();
-        const potentialMoves = piece.possibleMoves(board);
-        board.markPossibleMoves(potentialMoves);
-        //Register the event listener for handling when the marked square is selected
-        const handleInnerClick = () => {
-            if (isMarkedSquare(squareString)) {
-                const [row, col] = squareStringConverter(squareString);
-                activePiece.movePiece(row, col, board);
-                colorAllSquares();
-                    //Now that piece has moved, make the check for if opposing player is in check/checkmate:
-                    const enemyPlayerID = (currentPlayer === 0) ? 1 : 0;
-                    const kingPiece = getKingPiece(enemyPlayerID);
-                    if (kingPiece) {
-                        if (detectCheck(kingPiece, board)) {
-                            console.log("The opposing King has been placed in check.");
-                            if (detectCheckmate(kingPiece, board)) {
-                                console.log("The opposing King is in checkmate, game over.")
-                            } else {
-                                currentPlayer = (currentPlayer === 0) ? 1 : 0; //switch turns if opposing King isn't in checkmate. 
-                            }
-                        }
-                    }
-            }
-        };
-        const element = document.getElementById(squareString);
-        element.removeEventListener("click", handleInnerClick); 
-        element.addEventListener("click", handleInnerClick);
+
+function handleSquareClick(squareString) {
+    const piece = getPiece(squareString); //get the Piece on the square 
+    if (piece && piece.getPlayerId() == currentPlayer) { //if piece belongs to current player 
+        activePiece = piece; //sets activePiece to piece 
+        colorAllSquares(); //resets the colors of the squares 
+        const potentialMoves = piece.possibleMoves(board.board); //potential moves that the piece can go
+        board.markPossibleMoves(potentialMoves); //Marking the possible squares a piece can move to 
+        //Adding move piece functionality here. 
+        const markedSquares = document.getElementsByClassName("markedSquare");
+        for (let i = 0; i < markedSquares.length; i++) {
+            markedSquares[i].addEventListener("click", handleMoveToSquare);
```

Comment:
The event listener that is added to the marked squares on line `73` is never removed, which results in pieces being able to move to the marked squares after they are no longer marked.